system/Pager/PagerInterface.php

```diff
@@ -111,6 +111,17 @@ public function setPath(string $path, string $group = 'default');
 
 	//--------------------------------------------------------------------
 
+	/**
+	 * Returns the total number of items in data store.
+	 *
+	 * @param string $group
+	 *
+	 * @return integer
+	 */
+	public function getTotal(string $group = 'default'): int;
```

Comment:
Adding this to the interface would be breaking to those already implementing their own Pager via this interface.

Reply:
Can you give me example related what is breaking change for this. Thanks

Reply:
For example, in v4.0.4 I have implemented my own Pager.

```php
class MyPager implements PagerInterface
{
    // my code logic where I have implemented the interface's existing methods
}
```

Now, on upgrade to a later version, I would still be expecting this to work. Instead, I would get a `Fatal error` since I have not implemented the new method `getTotal`.

Reply:
Adding a new method to the interface means that this method has to be implemented in every class that uses this interface. If someone wrote his own pager class then after update this change will break his code.

Reply:
@paulbalandan @michalsn yes, will be breaking change if implement own Pager, but getTotal() is nice for spesific to get total instead of using getDetails()['total']

Example im implement [Fractal PaginatorInterface](https://github.com/thephpleague/fractal/blob/c1042fd31fd33d86fea7a3a54f643ad08905dc96/src/Pagination/PaginatorInterface.php#L19) will be need spesific get total. Here is my own implement:
```php
<?php

namespace App\Transformers;

use CodeIgniter\Pager\PagerInterface;
use League\Fractal\Pagination\PaginatorInterface;

class CodeIgniterPaginatorAdapter implements PaginatorInterface
{
    /**
     * The pagiantor.
     * 
     * @var \CodeIgniter\Pager\PagerInterface
     */
    protected $paginator;
    
    /**
     * Create a new illuminate pagination adapter.
     *
     * @param \CodeIgniter\Pager\PagerInterface $paginator
     * @return void
     */
    public function __construct(PagerInterface $paginator)
    {
        $this->paginator = $paginator;
    }

    /**
     * @inheritdoc
     */
    public function getCurrentPage()
    {
        return $this->paginator->getCurrentPage();
    }

    /**
     * @inheritdoc
     */
    public function getLastPage()
    {
        return $this->paginator->getLastPage();
    }

    /**
     * @inheritdoc
     */
    public function getTotal()
    {
        // instead of using
        // return $this->paginator->getDetails()['total'];
        
        return $this->paginator->getTotal();
    }

    /**
     * @inheritdoc
     */
    public function getCount()
    {
        return $this->paginator->getPageCount();
    }

    /**
     * @inheritdoc
     */
    public function getPerPage()
    {
        return $this->paginator->getPerPage();
    }

    /**
     * @inheritdoc
     */
    public function getUrl($page)
    {
        return $this->paginator->getPageURI($page);
    }

    /**
     * Get the paginator instance.
     * 
     * @return \CodeIgniter\Pager\Pager
     */
    public function getPaginator()
    {
        return $this->paginator;
    }
}
```

Reply:
Yes. I know what you are trying to achieve here. It would be a shortcut to what you have said. But we cannot discount the possibility that there is a developer out there who have implemented their own pager class. Why? Because it is part of the public API. *That developer* would be cursing on us once his perfectly working application will get errors after he made an update, considering that this is not a BC-breaking version update.

Reply:
I'm not against adding `getTotal()` to the `Pager` class but we should skip adding this method to the `PagerInterface` simply because this type of change is not worth introducing a BC.